flexport/latitude
# LDS-539 Add Transition Animations to SegmentedControl, LDS-593 Add Highlighter Component

## Description:
Adds `Highlighter` component which highlights a given element ref, updates `SegmentedControl` to use this component for selection animations. 

## Sandbox server:
https://core-63.sandbox.eng-dev.flexport.com/

## Jira:
[LDS-593](https://flexport.atlassian.net/browse/LDS-593)
[LDS-539](https://flexport.atlassian.net/browse/LDS-539)

## Manual Testing Plan:
*SegmentedControl*:
1. Run Storybook.
2. See SegmentedControl under the 'Navigation' section.
3. Observe new transition animation when selection buttons within the SegmentedControl. See video below.

*Highlighter*:
1. Run storybook.
2. See Highlighter under the 'Interaction' section.
3. Confirm Highlighter works as expected with knobs properly working.
4. Visit the SegmentedControl docs. (https://www-63.sandbox.eng-dev.flexport.com/design/components/Highlighter)
5. Confirm documentation is accurate.
6. Confirm live demo works correctly. 

## Screenshots:
![Kapture 2019-12-18 at 19 18 52](https://user-images.githubusercontent.com/7979583/71143666-43a4a800-21d0-11ea-821b-ed87315483af.gif)
![Kapture 2019-12-18 at 19 21 55](https://user-images.githubusercontent.com/7979583/71142749-2ae6c300-21cd-11ea-81ee-4704b84239a2.gif)
![Kapture 2019-12-18 at 19 23 04](https://user-images.githubusercontent.com/7979583/71142756-30dca400-21cd-11ea-9684-ba126b1d8040.gif)
![Kapture 2019-12-18 at 19 24 09](https://user-images.githubusercontent.com/7979583/71142764-34702b00-21cd-11ea-8744-f9c8abe797f2.gif)
